source/robot_model/tests/testRobotModel.cpp

```diff
@@ -1,8 +1,103 @@
 #include "robot_model/Model.hpp"
+
 #include <gtest/gtest.h>
 
+#include "robot_model/Exceptions/InvalidJointStateSizeException.hpp"
+#include "robot_model/Exceptions/FrameNotFoundException.hpp"
+
 using namespace RobotModel;
 
+class RobotModelTest : public testing::Test {
+protected:
+  void SetUp() override {
+    robot_name = "franka";
+    urdf = "/tmp/control_lib/robot_model/tests/panda_arm.urdf";
+    franka = Model(robot_name, urdf);
+    joint_state = StateRepresentation::JointState(robot_name, 7);
+  }
+
+  Model empty = Model();
+  Model franka;
+  std::string robot_name;
+  std::string urdf;
+
+  StateRepresentation::JointState joint_state;
+};
+
+TEST_F(RobotModelTest, TestInitEmptyModel) {
+  bool except_thrown = false;
+  try {
+    empty.init_model();
+  } catch (const std::invalid_argument& e) {
+    except_thrown = true;
+  }
+  EXPECT_TRUE(except_thrown);
+}
+
+TEST_F(RobotModelTest, TestSetName) {
+  empty.set_robot_name(robot_name);
+  EXPECT_EQ(empty.get_robot_name(), robot_name);
+}
+
+TEST_F(RobotModelTest, TestSetUrdfPath) {
+  empty.set_urdf_path(urdf);
+  EXPECT_EQ(empty.get_urdf_path(), urdf);
+}
+
+TEST_F(RobotModelTest, TestInitModel) {
+  empty.set_robot_name(robot_name);
+  empty.set_urdf_path(urdf);
+  EXPECT_NO_THROW(empty.init_model());
+}
+
+TEST_F(RobotModelTest, TestConstructor) {
+  Model tmp(robot_name, urdf);
```

Comment:
Here I get a warning ` [OsqpEigen::Data::setHessianMatrix] The hessian matrix was already set. Please use clearHessianMatrix() method to deallocate memory.` due to the operation `franka=tmp`

Reply:
This would be a fair place to add `clearHessianMatrix()` to the appropriate constructor, since you've correctly and neatly identified an issue as part of the test case. Since it's a small fix I think it can go with this PR.